idunn/api/instant_answer.py

```diff
@@ -223,34 +229,62 @@ async def get_instant_answer(
     if lang in nlu_allowed_languages:
         try:
             intentions = await nlu_client.get_intentions(
-                normalized_query, lang, extra_geocoder_params
+                normalized_query,
+                lang,
+                extra_geocoder_params,
+                allow_types=[
+                    IntentionType.BRAND,
+                    IntentionType.CATEGORY,
+                    IntentionType.POI,
+                    IntentionType.ANY_PLACE,
+                ],
             )
-            if intentions:
+            if intentions and intentions[0].type in [
+                IntentionType.BRAND,
+                IntentionType.CATEGORY,
+            ]:
                 return await get_instant_answer_intention(intentions[0], query=q, lang=lang)
         except NluClientException:
             # No intention could be interpreted from query
-            pass
+            intentions = []
 
     # Direct geocoding query
     query = QueryParams.build(q=normalized_query, lang=lang, limit=5, **extra_geocoder_params)
-    bragi_response = await bragi_client.autocomplete(query)
-    geocodings = (feature["properties"]["geocoding"] for feature in bragi_response["features"])
-    geocodings = sorted(
-        (
-            (
-                result_filter.rank_bragi_response(normalized_query, feature),
-                feature,
-            )
-            for feature in geocodings
-            if result_filter.check_bragi_response(normalized_query, feature)
-        ),
-        key=lambda item: -item[0],  # sort by descending rank
+
+    async def fetch_pj_response():
+        if not (settings["IA_CALL_PJ_POI"] and user_country == "fr" and intentions):
+            return []
+
+        return await run_in_threadpool(
+            pj_source.search_places,
+            normalized_query,
+            intentions.place_in_query,
```

Comment:
`intentions` is a list, right?  
```suggestion
            intentions[0].place_in_query,
```